### PR TITLE
Fix username regex

### PR DESCRIPTION
**Pull request by wunter8**
Now you can include a "+" in your username (for example, in an email alias).

I also added some code to convert the user/types.go ErrInvalidArgument into an errHTTPBadRequestInvalidArgument so that the error message in the web UI says "invalid request: invalid argument" instead of "internal server error."

I feel like that part of the PR isn't super clean, so you might want to rewrite it (all 6 lines)

### CODE REVIEW

File: server/server_account.go [hunk 1]
```diff
@@ -37,7 +37,11 @@ func (s *Server) handleAccountCreate(w http.ResponseWriter, r *http.Request, v *
 	}
 	logvr(v, r).Tag(tagAccount).Field("user_name", newAccount.Username).Info("Creating user %s", newAccount.Username)
 	if err := s.userManager.AddUser(newAccount.Username, newAccount.Password, user.RoleUser); err != nil {
-		return err
+		if err.Error() == "invalid argument" {
```
Comment (binwiederhier) [inline server/server_account.go:40]:
```suggestion
		if errors.Is(err, ErrInvalidArgument) {
		  return errHTTPBadRequestInvalidUsername
		}
		return err	
```

File: server/errors.go [hunk 2]
```diff
@@ -122,6 +122,7 @@ var (
 	errHTTPBadRequestTemplateInvalid                 = &errHTTP{40043, http.StatusBadRequest, "invalid request: could not parse template", "https://ntfy.sh/docs/publish/#message-templating", nil}
 	errHTTPBadRequestTemplateDisallowedFunctionCalls = &errHTTP{40044, http.StatusBadRequest, "invalid request: template contains disallowed function calls, e.g. template, call, or define", "https://ntfy.sh/docs/publish/#message-templating", nil}
 	errHTTPBadRequestTemplateExecuteFailed           = &errHTTP{40045, http.StatusBadRequest, "invalid request: template execution failed", "https://ntfy.sh/docs/publish/#message-templating", nil}
+	errHTTPBadRequestInvalidArgument                 = &errHTTP{40046, http.StatusBadRequest, "invalid request: invalid argument", "", nil}
```
Comment (binwiederhier) [inline server/errors.go:125]: errHTTPBadRequestInvalidUsername?

File: user/types.go [hunk 3]
```diff
@@ -241,7 +241,7 @@ const (
 )
 
 var (
-	allowedUsernameRegex     = regexp.MustCompile(`^[-_.@a-zA-Z0-9]+$`)     // Does not include Everyone (*)
+	allowedUsernameRegex     = regexp.MustCompile(`^[-_.+@a-zA-Z0-9]+$`)    // Does not include Everyone (*)
```
Comment (binwiederhier) [inline user/types.go:244]: Can you check if there is a regex in the frontend too somewhere?

**Review by binwiederhier — APPROVED** [review]

**Comment by wunter8** [comment]
I'm planning to get to this today

File: server/errors.go (hunk 2 above)
Comment (wunter8) [reply server/errors.go:125]: Right now, 'ErrInvalidArgument' is returned for invalid usernames, topics, roles, and tiers.

But, in this section of the code specifically, we're only checking for usernames, so the 'ErrInvalidArgument' will always mean 'errHTTPBadRequestInvalidUsername'

File: user/types.go (hunk 3 above)
Comment (wunter8) [reply user/types.go:244]: I looked and didn't find anything

**Pull request merged**